contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthSchema.java

```diff
@@ -0,0 +1,388 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.contrib.parser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * This is schema that defines fields and their constraints for fixed width files
+ * The operators use this information to validate the incoming tuples.
+ * Information from JSON schema is saved in this object and is used by the
+ * operators
+ * <p>
+ * <br>
+ * <br>
+ * Example schema <br>
+ * <br>
+ * {{ "padding": " ","alignment"="left","fields": [ {"name": "adId",
+ * "type": "Integer","padding":"0", "fieldLength": 3} , { "name": "adName",
+ * "type": "String", "alignment": "right","fieldLength": 20}, { "name": "bidPrice", "type":
+ * "Double", "fieldLength": 5}, { "name": "startDate", "type": "Date", "fieldLength": 10,
+ * "format":"dd/MM/yyyy" }, { "name": "securityCode", "type": "Long","fieldLength": 10 },
+ * { "name": "active", "type":"Boolean","fieldLength": 2} ] }}
+ */
+public class FixedWidthSchema extends Schema
+{
+
+  /**
+   * JSON key string for record length
+   */
+  public static final String FIELD_LEN = "length";
+  /**
+   * JSON key string for Global Padding Character
+   */
+  public static final String GLOBAL_PADDING_CHARACTER = "padding";
```

Comment:
Why not have a single variable: PADDING, for both global and field as the name is the same?